USE_CASES.md

```diff
@@ -127,3 +128,291 @@ Find more information about all of SendGrid's whitelabeling related documentatio
 You can find documentation for how to view your email statistics via the UI [here](https://app.sendgrid.com/statistics) and via API [here](https://github.com/sendgrid/sendgrid-ruby/blob/master/USAGE.md#stats).
 
 Alternatively, we can post events to a URL of your choice via our [Event Webhook](https://sendgrid.com/docs/API_Reference/Webhooks/event.html) about events that occur as SendGrid processes your email.
+
+<a name="heroku_tutorial"></a>
+# Heroku Tutorial
+
+### Create a Heroku Account
+
+#1
+
+If you haven't already, then go ahead and Sign up for a free [Heroku](https://www.heroku.com/) account, and install the [Heroku toolbelt](https://devcenter.heroku.com/articles/heroku-cli) for your operating system.
+
+#2
+
+Once, you have a Heroku account, add your credit card info to your account. **Your card will not be charged for free plans** but this validation is needed in order to prevent spammers from abusing the free SendGrid plan.
+
+#3
+
+If you already have a Heroku account that you are signed into through your command line utility:
+
+re-establish your credentials by: `$ heroku auth:logout`
+
+...and then log back in: `$ heroku login`
+
+If you are new to Heroku:
+
+Open your command line utility and login: `$ heroku login`
+
+### Create a Rails App
+
+#4
+
+The first thing you'll want to do is create a new Rails application by running the `rails new` command.
+
+Lets, call the app, 'hello'
+
+```ruby
+$ rails new hello
+```
+
+### Create Heorku Application
+
+#5
+
+`cd` into the `hello` directory and run:
+
+```
+$ heroku create
+
+Creating app... done, ⬢ morning-stream-10053
+https://morning-stream-10053.herokuapp.com/ | https://git.heroku.com/morning-stream-10053.git
+```
+
+This creates a randomly generated application name and adds heroku as one of your remote destinations so you can push your repo to Heroku easily. If you want a custom name for your application, see Heroku's documentation [here](https://devcenter.heroku.com/articles/creating-apps#creating-a-named-app).
+
+### Replace SQLite with PostgreSQL
+
+#6
+
+If you are using SQLite it must be replaced with PostgreSQL because Heroku does not support `sqlite3`. [Follow these directions](https://devcenter.heroku.com/articles/sqlite3).
+
+### Test Deployment to Heroku
+
+#7
+
+If the setup has been done correctly, you can now try to deploy your application to Heroku
+
+```
+$ git push heroku master
+```
+
+### Install the SendGrid add-on from the command line:
+
+#8
+
+Lets add SendGrid to our app:
+
+```
+$ heroku addons:create sendgrid:starter
+
+Creating sendgrid:starter on ⬢ morning-stream-10053... free
+Use heroku addons:docs sendgrid to view documentation
+```
+#9
+
+Verify that you installed SendGrid by typing the following on the command line:
+
+```
+$ heroku addons
```

Comment:
Might be useful to include sample output here, the same as with most of the other `heroku` commands in this tutorial.